# Hausaufgabe 1
## Lärmbelastung am Arbeitsplatz
<!-- Lizensiert unter (CC BY 2.0) Gert Herold, 2019 -->

Für die Beurteilung der tagsüber auftretenden Lärmbelastung in einem Gewerbegebiet wurden über mehrere Monate an Werktagen der Mittelungspegel $L_\text{A,eq}$ an einem Immissionsort bestimmt.


In [1]:
LAeq = {'Januar': [65.6473, 63.6885, 63.3891, 66.0919, 64.1625, 66.4128, 61.4494, 50.8258, 68.1941, 73.9349, 65.2068, 
                   62.4992, 62.3178, 62.4047, 63.087, 57.3899, 60.4145, 58.0149, 64.244, 61.5168, 65.4761, 62.1265], 
        'Februar': [67.9317, 64.0677, 73.9535, 53.5182, 59.7654, 67.5074, 75.6416, 61.7568, 63.2183, 61.8684, 
                    69.6573, 61.5635, 66.4003, 61.401, 56.6372, 64.5677, 65.5159, 69.4661, 62.4478, 59.9132], 
        'März': [65.8138, 64.2037, 64.4033, 62.6344, 68.8017, 64.8475, 72.5233, 68.5553, 66.2952, 54.8628, 
                 66.0116, 65.1014, 67.0548, 68.2222, 60.9956, 67.12, 60.1885, 72.7744, 56.3402, 54.1966], 
        'April': [54.7464, 58.5472, 57.4044, 72.7804, 61.3675, 56.2866, 68.5719, 60.0674, 56.8157, 67.3792, 
                  66.1168, 60.8252, 70.0377, 63.6455, 71.0847, 65.5137, 70.794, 63.547, 56.9013, 75.2468], 
        'Mai': [60.2711, 62.0058, 59.498, 61.983, 64.2133, 64.0092, 66.3051, 71.9608, 62.3977, 56.8344, 57.0884, 
                59.6712, 54.629, 67.1251, 60.5089, 61.4451, 62.9905, 56.0169, 58.6934, 66.3736, 66.0927], 
        'Juni': [60.7841, 72.0527, 56.4714, 61.2751, 61.8458, 49.0346, 72.6876, 64.8317, 57.7771, 73.2559, 
                 65.9283, 65.1476, 59.965, 63.5311, 55.3716, 66.9751, 61.1278, 63.6702, 69.0103]}

In [11]:
print(type('Mai'))
print(type('Mai'[0]))
#LAeq.keys(),LAeq.values()

<class 'str'>
<class 'str'>


In [2]:
for i,j in LAeq.items():
    a=max(j)
    print("Im {} wurde ein maximaler Mittelungspegel von {:.1f} dB gemessen.".format(i,a))
    

Im Januar wurde ein maximaler Mittelungspegel von 73.9 dB gemessen.
Im Februar wurde ein maximaler Mittelungspegel von 75.6 dB gemessen.
Im März wurde ein maximaler Mittelungspegel von 72.8 dB gemessen.
Im April wurde ein maximaler Mittelungspegel von 75.2 dB gemessen.
Im Mai wurde ein maximaler Mittelungspegel von 72.0 dB gemessen.
Im Juni wurde ein maximaler Mittelungspegel von 73.3 dB gemessen.


**1. Geben Sie für jeden Monat den jeweils höchsten Pegel mit einer Nachkommastelle als Fließtext an.**
 
Beispiel:

    Im August wurde ein maximaler Mittelungspegel von 75.2 dB gemessen.
    
*(Für einen Hinweis zur Textformatierung unter Python den ausgeblendeten Abschnitt öffnen)*

In [26]:
for A in LAeq.keys():
    for a in LAeq.values():
        a_max=max(a)
    print('Im %s wurde ein maximaler Mittelungspegel von %.1f dB gemessen.'%(A,a_max))

Im Januar wurde ein maximaler Mittelungspegel von 73.3 dB gemessen.
Im Februar wurde ein maximaler Mittelungspegel von 73.3 dB gemessen.
Im März wurde ein maximaler Mittelungspegel von 73.3 dB gemessen.
Im April wurde ein maximaler Mittelungspegel von 73.3 dB gemessen.
Im Mai wurde ein maximaler Mittelungspegel von 73.3 dB gemessen.
Im Juni wurde ein maximaler Mittelungspegel von 73.3 dB gemessen.


In [3]:
#Hinweis: Zeichketten in Python mit formatierenden Platzhaltern -- verschiedene Wege zum gleichen Ziel
s = 'Sekunde'
m = 'Minute'
h = 'Stunde'
s_per_m = 60
h_per_m = 1/60

print('%04d: Eine %s hat %d %sn. Das sind %8.4f %sn. Eine %s entspricht demnach %.6f %sn.' % (1, m, s_per_m, s, h_per_m, h, s, h_per_m/s_per_m, h))
print('{:04d}: Eine {} hat {} {}n. Das sind {:8.4f} {}n. Eine {} entspricht demnach {:.6f} {}n.'.format(2, m, s_per_m, s, h_per_m, h, s, h_per_m/s_per_m, h))
print(f'{3:04d}: Eine {m} hat {s_per_m} {s}n. Das sind {h_per_m:8.4f} {h}n. Eine {s} entspricht demnach {h_per_m/s_per_m:.6f} {h}n.')

0001: Eine Minute hat 60 Sekunden. Das sind   0.0167 Stunden. Eine Sekunde entspricht demnach 0.000278 Stunden.
0002: Eine Minute hat 60 Sekunden. Das sind   0.0167 Stunden. Eine Sekunde entspricht demnach 0.000278 Stunden.
0003: Eine Minute hat 60 Sekunden. Das sind   0.0167 Stunden. Eine Sekunde entspricht demnach 0.000278 Stunden.


In [4]:
for A,a in LAeq.items():
    a_max=max(a)
    print('im {} wurde ein maximaler mittelungspegel von {:.1f} dB gemessen.' .format(A,a_max) )
    

im Januar wurde ein maximaler mittelungspegel von 73.9 dB gemessen.
im Februar wurde ein maximaler mittelungspegel von 75.6 dB gemessen.
im März wurde ein maximaler mittelungspegel von 72.8 dB gemessen.
im April wurde ein maximaler mittelungspegel von 75.2 dB gemessen.
im Mai wurde ein maximaler mittelungspegel von 72.0 dB gemessen.
im Juni wurde ein maximaler mittelungspegel von 73.3 dB gemessen.


**2. Berechnen Sie für jeden Monat die Anzahl der Tage, an denen der zulässige Richtwert von 65 dB überschritten wurde und berechnen Sie deren prozentualen Anteil an der Gesamtzahl der Werktage im jeweiligen Monat.**

Beispiel:

    Juli    :  6 Tage (26.1 %)
    August  :  9 Tage (40.9 %)

<!-- Werktage Juli:23, August: 22 -->

In [11]:
n=0
for a,b in LAeq.items():
    m=len(b)
    for j in b:
        if j>65:
            n+=1
    print("{}:{} tage({:.1f%%})".format(i,n,n*100/m))

ValueError: Invalid format specifier

In [12]:
for A,a in LAeq.items():
    n=0
    m=len(a)
    for i in a:
         if i>65:
                n+=1
    print("%s:%s Tage(%.1f%%)"%(A,n,n*100/m))

Januar:7 Tage(31.8%)
Februar:8 Tage(40.0%)
März:11 Tage(55.0%)
April:9 Tage(45.0%)
Mai:5 Tage(23.8%)
Juni:7 Tage(36.8%)


In [14]:
for A,a in LAeq.items():
    z=0
    n=len(a)
    for j in a:
        if j>65:
            z=z+1
         
            
    print('%s:%d Tage(%.1f%%)' %(A,z,(100*z/n)))

Januar:7 Tage(31.8%)
Februar:8 Tage(40.0%)
März:11 Tage(55.0%)
April:9 Tage(45.0%)
Mai:5 Tage(23.8%)
Juni:7 Tage(36.8%)


In [6]:
def tage():
    for A,a in LAeq.items():
        z=0
        for x in a:
            if x>65:
                z=z+1
                n=len(a)
        print('%s:%d Tage(%.1f%%)' %(A,z,(100*z/n)))
    return

tage()

#print('%s:%d Tage(%.1f%%)' %(A,z,(100*z/n)))

Januar:7 Tage(31.8%)
Februar:8 Tage(40.0%)
März:11 Tage(55.0%)
April:9 Tage(45.0%)
Mai:5 Tage(23.8%)
Juni:7 Tage(36.8%)


**3. Durch intensive Lärmminderungsmaßnahmen könnte der Pegel um durchschnittlich 6 dB gesenkt werden. Schätzen Sie basierend auf den gegebenen Werten ab, wie viele (Werk-)Tage im Jahr es noch immer zu Richtwertüberschreitungen kommen würde. Wie sieht es bei 7 dB durchschnittlicher Pegelverringerung aus?**

In [29]:
for A,a in LAeq.items():
    z=0
    n=len(a)
    for j in a:
        if (j-6)>65:
            z=z+1
            
           
    print('%s:%d Tage(%.1f%%)' %(A,z,(100*z/n)))

Januar:1 Tage(4.5%)
Februar:2 Tage(10.0%)
März:2 Tage(10.0%)
April:3 Tage(15.0%)
Mai:1 Tage(4.8%)
Juni:3 Tage(15.8%)


In [30]:
z=0
def tage(z):
    for A,a in LAeq.items():
        #n=len(a)
        for x in a:
            if (x-6)>65:
                z=z+1
    return z

t=tage(z)*2

print('Es würde noch immer %s Tage im Jahr zu Richtwertüberschreitungen kommen ' %t)
#print(tage(z)*2)

Es würde noch immer 24 Tage im Jahr zu Richtwertüberschreitungen kommen 


In [10]:
# Hier eigenen Code schreiben ...
z=0
def tage(z):
    for A,a in LAeq.items():
        #n=len(a)
        for x in a:
            if (x-7)>65:
                z=z+1
    return z

t_1=tage(z)*2
print('Es würde noch immer %s Tage im Jahr zu Richtwertüberschreitungen kommen ' %t_1)
#print(tage(z)*2)



Es würde noch immer 20 Tage im Jahr zu Richtwertüberschreitungen kommen 
